playback/src/audio_backend/rodio.rs

```diff
@@ -0,0 +1,96 @@
+use super::{Open, Sink};
+extern crate rodio;
+use std::{io, thread, time};
+use std::process::exit;
+
+pub struct RodioSink {
+    rodio_sink: rodio::Sink,
+}
+
+fn list_outputs() {
+    println!("Default Audio Device:\n  {:?}", rodio::default_output_device().map(|e| e.name()));
```

Comment:
iirc we don't use `println` anywhere in librespot. Instead `info!` and `debug!` are used. Please change each instance into the appropriate one of the aforementioned. I thikn pretty much all of them should be `debug!`, but feel free to disagree.

Reply:
This function is used as the result of `librespot --device ?` which is itself similar to `librespot --backend ?` which [uses `println!`](https://github.com/librespot-org/librespot/blob/master/src/main.rs#L79). It's a CLI feature rather than a log message. So it's definitely not `debug!` and I feel like the prefix would make it rather messy.

That being said the logging of the formats was really a debug thing. I was intending to switch that to `debug!` but forgot. I'll see how it looks like that but I guess the other option is to conditionally compile the format printing stuff.

This is how it looks atm:
```
>target\release\librespot.exe -n hi --backend ?
INFO:librespot: librespot 518e9eb (2019-03-10). Built on 2019-03-10. Build ID: hiaBYkqz
Available Backends :
- rodio (default)
- pipe

>target\release\librespot.exe -n hi --device ?
INFO:librespot: librespot 518e9eb (2019-03-10). Built on 2019-03-10. Build ID: hiaBYkqz
INFO:librespot_playback::audio_backend::rodio: Using rodio sink
Default Audio Device:
  Some("Speakers (Music Streamer III)")
Available Audio Devices:
- 2769M (NVIDIA High Definition Audio)
  Default format:
    Format { channels: 2, sample_rate: SampleRate(48000), data_type: F32 }
  All formats:
    SupportedFormat { channels: 2, min_sample_rate: SampleRate(48000), max_sample_rate: SampleRate(48000), data_type: F32 }
- Speakers (Music Streamer III)
  Default format:
    Format { channels: 2, sample_rate: SampleRate(48000), data_type: F32 }
  All formats:
    SupportedFormat { channels: 2, min_sample_rate: SampleRate(48000), max_sample_rate: SampleRate(48000), data_type: F32 }
- CABLE Input (VB-Audio Virtual Cable)
  Default format:
    Format { channels: 2, sample_rate: SampleRate(96000), data_type: F32 }
  All formats:
    SupportedFormat { channels: 2, min_sample_rate: SampleRate(96000), max_sample_rate: SampleRate(96000), data_type: F32 }
- Realtek Digital Output (Realtek High Definition Audio)
  Default format:
    Format { channels: 2, sample_rate: SampleRate(192000), data_type: F32 }
  All formats:
    SupportedFormat { channels: 2, min_sample_rate: SampleRate(192000), max_sample_rate: SampleRate(192000), data_type: F32 }
```

Reply:
hmm, good point, missed that it was output in response to a user input, makes sense to use `println!` I think the formats can use `debug!` though. This also answers comment below.

Reply:
also, we can probably lose the `Some` in `Some("Speakers (Music Streamer III)")` for the default audio device.

Reply:
```
INFO:librespot: librespot 518e9eb (2019-03-10). Built on 2019-03-10. Build ID: Hey1wefj
DEBUG:librespot_playback::audio_backend::rodio: Using rodio sink
Default Audio Device:
  Speakers (Music Streamer III)
DEBUG:librespot_playback::audio_backend::rodio:   Available formats:
DEBUG:librespot_playback::audio_backend::rodio:     (default) 2ch, F32, min SampleRate(48000), max SampleRate(48000)
Other Available Audio Devices:
  2769M (NVIDIA High Definition Audio)
DEBUG:librespot_playback::audio_backend::rodio:   Available formats:
DEBUG:librespot_playback::audio_backend::rodio:     (default) 2ch, F32, min SampleRate(48000), max SampleRate(48000)
  CABLE Input (VB-Audio Virtual Cable)
DEBUG:librespot_playback::audio_backend::rodio:   Available formats:
DEBUG:librespot_playback::audio_backend::rodio:     (default) 2ch, F32, min SampleRate(96000), max SampleRate(96000)
  Realtek Digital Output (Realtek High Definition Audio)
DEBUG:librespot_playback::audio_backend::rodio:   Available formats:
DEBUG:librespot_playback::audio_backend::rodio:     (default) 2ch, F32, min SampleRate(192000), max SampleRate(192000)
```
Is the new output... so for most people:
```
INFO:librespot: librespot 518e9eb (2019-03-10). Built on 2019-03-10. Build ID: Hey1wefj
Default Audio Device:
  Speakers (Music Streamer III)
Other Available Audio Devices:
  2769M (NVIDIA High Definition Audio)
  CABLE Input (VB-Audio Virtual Cable)
  Realtek Digital Output (Realtek High Definition Audio)
```